bump_dictionary/utils.py

```diff
@@ -0,0 +1,129 @@
+import json
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Type
+
+import typer
+from jsonschema import Draft202012Validator
+from pydantic import ValidationError
+
+from .logger import log_error, logger
+from .models.legacy_dictionary_model import (
+    CategoricalNeurobagel,
+    ContinuousNeurobagel,
+    IdentifierNeurobagel,
+    Neurobagel,
+    ToolNeurobagel,
+)
+
+
+def load_json(file: Path) -> Any:
+    """Load a JSON file and return its content if file has valid encoding and is valid JSON."""
+    try:
+        with open(file, "r", encoding="utf-8") as f:
+            return json.load(f)
+    except UnicodeDecodeError:
+        log_error(
+            logger,
+            f"Data dictionary must have UTF-8 encoding: {file}. "
+            "[italic]TIP: Need help converting your file? Try a tool like iconv (http://linux.die.net/man/1/iconv) or https://www.freeformatter.com/convert-file-encoding.html.[/italic]",
+        )
+    except json.JSONDecodeError:
+        log_error(
+            logger,
+            f"Data dictionary is not valid JSON: {file}.",
+        )
+
+
+def check_overwrite(file: Path, overwrite: bool):
+    """Exit program gracefully if an output file already exists but --overwrite is not enabled."""
+    if file.exists() and not overwrite:
+        raise typer.Exit(
+            typer.style(
+                f"Output file {file} already exists. Use --overwrite or -f to overwrite.",
+                fg=typer.colors.RED,
+            )
+        )
```

Comment:
why not return a boolean here and then handle the exiting in the main method? That would also avoid this return-less side effect and makes the main method more readable imho